source/develop/release-management/features/virt/vm-lifecycle-with-kubevirt.md

```diff
@@ -0,0 +1,130 @@
+---
+title: VM lifecycle in Kubevirt 
+category: feature
+authors: arik
+wiki_category: Feature
+wiki_title: Features/VM lifecycle in Kubevirt
+wiki_revision_count: 20
+wiki_last_updated: 2015-10-21
+feature_name: VM lifecycle in Kubevirt
+feature_modules: engine
+feature_status: 
+---
+
+This page proposes a gradual way to integrate the virtualization management platform oVirt and containers management platform Kubernetes.
+
+
+# Motivation
+
+The virtualization management platform oVirt and the containers management platform Kubernetes share common capabilities. For example, both platforms need to determine on which node to run their managed entity (VM/container), run and then monitor it.  
+
+It is thus tempting to reuse the common parts of these platforms instead of developing and maintaining a dedicated infratructure in each of them.  
+
+However, it is not as easy as it may sound. The differences between virtual machines and containers require different capabilities and decision making by the management platform. Moreover, these two platforms are written in different technology and designed differently.  
+
+Therefore, it only makes sense to consider a gradual integration between the two platforms by delegating stuff from oVirt into Kubevirt, an extended form of Kubernetes that includes components that are specific to virtual machines. A natural choice would be to start with delegating the scheduling and virtual machine creation from oVirt to Kubernetes.  
+
+# The Current Process in oVirt
+
+In this section I elaborates on the process of running a virtual machine in oVirt. The creation of the virtual machine is done as a separate operation before it is being run so we can assume that the VM already exists in the database at this stage.  
+
+The process looks as follow:  
+0. Optional: a request is sent from client (UI/REST-API).  
+1. Instance of RunVmCommand is created.  
+2. The input is being verified. Specifically, the scheduler validates that there is a host capable of running the VM.  
+3. The engine retrieves all the properties of the VM to run.  
+4. The scheduler chooses a host to run the VM on.  
+5. The engine locks host devices, plug lun disks and update cinder disks.  
+6. The engine locks the monitoring thread.  
+7. The engine sends a request to VDSM on the chosen host to run the VM.  
+8. If the response if positive, the engine set the VM to WaitForLaunch state (and few other fields).  
+9. The monitoring lock is released.  
+10. The monitoring receives the VM as running (positive flow) or down (negative flow). If the VM fails to run, the engine tries to run it few more time (by default 2 additional attempts).
+
+# Proposed Approach
+
+So this proposal split the integration between oVirt and Kubernetes in regards to VM lifecycle into several phases. Note that some of them can be done in parallel.   
+
+## Milestone 1: Basic integration
+
+### Goal
+1. Add Kubevirt into oVirt.  
```

Comment:
I understand this line as "add kubevirt to the environment in which oVirt runs" do you mean it this way or do you mean something else?